esp-hal/src/lcd_cam/lcd/i8080.rs

```diff
@@ -232,22 +232,25 @@ impl<'d, DM: Mode> I8080<'d, DM> {
     }
 
     /// Associates a CS pin with the I8080 interface.
-    pub fn with_cs<CS: PeripheralOutput>(self, cs: impl Peripheral<P = CS> + 'd) -> Self {
+    pub fn with_cs(self, cs: impl Peripheral<P = impl Into<OutputConnection> + 'd> + 'd) -> Self {
```

Comment:
At this point I'm not sure if the peripheral ref pattern still make sense for gpios when used as signals. I'm thinking the `impl Peripheral` wrapper should just be skipped now.

Reply:
How else would you pass a `&mut io.pins.gpio4` and be able to get it back?

Reply:
Implement `Into<OutputConnection>` for `&mut GpioPin`?

Reply:
Why would I do that, when we get it for free?

Reply:
`impl Peripheral<P = impl Into<OutputConnection> + 'd> + 'd` just seems a like a mouthful. Was just hoping it could be shorter.

Reply:
> so that we can reduce the amount of trait bloat in the HAL.

😛

Reply:
`From<&mut GpioPin<N>>` would essentially make `OutputSignal` semi-`Clone` - if you have a GpioPin, you can make as many as you want. That is a potential issue using safe code.

Reply:
Ah that is true, tricky

Reply:
Further thoughts on the topic. I think I have a solution for the `Into`/`Clone` problem.

If we keep the `PeripheralInput` and `PeripheralOutput` traits, then we can implement `PeripheralOutput` for `&mut GpioPin` (or `BorrowMut<GpioPin>` if the trait solver doesn't mind).

Then constructors can be simplified to `pub fn with_cs(self, cs: impl PeripheralOuput)`

There's an interesting side effect here where you can implement `PeripheralInput` for `&GpioPin` (instead of `&mut`). Though since pins may need to have input enabled and/or alternative functions set, this may not be doable anyway.

Reply:
> If we keep the PeripheralInput and PeripheralOutput traits,

I'm considering undoing this change. `P = impl Into` has some unforseen difficulties. However, I'm not (yet) considering breaking the PeripheralRef pattern just in one case (as in, signals vs other peripheraly bits). Maybe we'll arrive there eventually, but that's a change that can and should be done separately.

I'll admit that the current state is very likely to undergo drastic changes until the PR becomes ready.